### ISSUE TITLE
AttributeProvider not working for TablesExtension and images

### ISSUE THREAD

**Issue by rmuller** [atlassian/commonmark-java#31]
As documented in the source code (`TableHtmlRenderer#renderBlock()`):

````
// TODO: What about attributes? If we got the renderer instead of the visitor, we could call getAttributes.
````

To be able to set custom attributes would be very handy especially when working with Tables!

More in general: IMO `AttributeProvider` should be working for all tags, including images.

**Comment by robinst** [comment]
Thanks for the report! Yeah, let's fix it properly and also make it work for images.

In the case of images, we currently don't support it because we use the same renderer to render the alt attribute (see [here](https://github.com/atlassian/commonmark-java/blob/master/commonmark/src/main/java/org/commonmark/html/HtmlRenderer.java#L310)). I'm thinking we should change that and create a new renderer, then take the rendered string for the alt value, put it in the attributes and then let the attribute provider extend it.

**Comment by robinst** [comment]
Are you interested in contributing this yourself? Otherwise I'll pick it up when I get to it :).

**Comment by robinst** [comment]
Fixed with #49.

**Comment by robinst** [comment]
Released in 0.5.0.
